src/reducers/users.tsx

```diff
@@ -65,7 +67,7 @@ export const requestLogOut = ({
 };
 
 export type UsersState = {
-  currentUser: User | null;
+  currentUser: User | null | undefined;
```

Comment:
Why not keeping `User | null` and have a boolean flag to determine the loading status instead?

Reply:
I'd want to change the return of `getCurrentUser` to `User | null` as well but I guess that would make sense

Reply:
Yes, this does make more sense since `undefined` is an implementation detail of the state

Reply:
So, we keep `| undefined` here to avoid the loading flag in the state, right?

Reply:
Whoops, when you first commented I mistook the context for the `profile: User | null` prop on `App` and `Navbar`.

> So, we keep `| undefined` here to avoid the loading flag in the state, right?

Yeah, I think it's OK to use `undefined` here for a loading state. When something is `null` it says to me that "it exists but it's empty." As you eluded to, storing a separate loading flag would make things more complex.

Reply:
thanks!